crates/bitcoind_rpc/src/lib.rs

```diff
@@ -0,0 +1,476 @@
+    ///
+    /// The `tx_filter` parameter takes in a closure that filters out irrelevant transactions so
+    /// they do not get included in the [`TxGraph`] update. We have provided two closures;
+    /// [`empty_filter`] and [`indexer_filter`] for this purpose.
+    ///
+    /// The `anchor_map` parameter takes in a closure that creates anchors of a specific type.
+    /// [`confirmation_height_anchor`] and [`confirmation_time_anchor`] are avaliable to create
+    /// updates with [`ConfirmationHeightAnchor`] and [`ConfirmationTimeAnchor`] respectively.
+    pub fn into_tx_graph_update<F, M, A>(self, tx_filter: F, anchor_map: M) -> TxGraph<A>
+    where
+        F: FnMut(&Transaction) -> bool,
+        M: Fn(&CheckPoint, &Block, usize) -> A,
+        A: Clone + Ord + PartialOrd,
+    {
+        match self {
+            EmittedUpdate::Block(e) => e.into_tx_graph_update(tx_filter, anchor_map),
+            EmittedUpdate::Mempool(e) => e.into_tx_graph_update(tx_filter),
+        }
+    }
+}
+
+/// An emitted block.
+#[derive(Debug, Clone)]
+pub struct EmittedBlock {
+    /// The checkpoint constructed from the block's height/hash and connected to the previous block.
+    pub cp: CheckPoint,
+    /// The actual block of the chain.
+    pub block: Block,
+}
+
+impl EmittedBlock {
+    /// Get the emission's checkpoint.
+    pub fn checkpoint(&self) -> CheckPoint {
+        self.cp.clone()
+    }
+
+    /// Transforms the emitted update into a [`TxGraph`] update.
+    ///
+    /// The `tx_filter` parameter takes in a closure that filters out irrelevant transactions so
+    /// they do not get included in the [`TxGraph`] update. We have provided two closures;
+    /// [`empty_filter`] and [`indexer_filter`] for this purpose.
+    ///
+    /// The `anchor_map` parameter takes in a closure that creates anchors of a specific type.
+    /// [`confirmation_height_anchor`] and [`confirmation_time_anchor`] are avaliable to create
+    /// updates with [`ConfirmationHeightAnchor`] and [`ConfirmationTimeAnchor`] respectively.
+    pub fn into_tx_graph_update<F, M, A>(self, mut tx_filter: F, anchor_map: M) -> TxGraph<A>
+    where
+        F: FnMut(&Transaction) -> bool,
+        M: Fn(&CheckPoint, &Block, usize) -> A,
+        A: Clone + Ord + PartialOrd,
+    {
+        let mut tx_graph = TxGraph::default();
+        let tx_iter = self
+            .block
+            .txdata
+            .iter()
+            .enumerate()
+            .filter(move |(_, tx)| tx_filter(tx));
+        for (tx_pos, tx) in tx_iter {
+            let txid = tx.txid();
+            let _ = tx_graph.insert_anchor(txid, anchor_map(&self.cp, &self.block, tx_pos));
+            let _ = tx_graph.insert_tx(tx.clone());
+        }
+        tx_graph
+    }
+}
+
+/// An emitted subset of mempool transactions.
+#[derive(Debug, Clone)]
+pub struct EmittedMempool {
+    /// The checkpoint of the last-seen tip.
+    pub cp: CheckPoint,
```

Comment:
please explain in the docs how I am meant to use this information.

Reply:
I removed `EmittedMempool::cp`. It's not useful.